app/models/search_builder.rb

```diff
@@ -1,4 +1,10 @@
 # frozen_string_literal: true
 class SearchBuilder < Blacklight::SearchBuilder
   include Blacklight::Solr::SearchBuilderBehavior
+  include Hydra::AccessControlsEnforcement
+
+  def add_access_controls_to_solr_params(*args)
+    return if current_ability.can?(:manage, Valkyrie::Model)
+    apply_gated_discovery(*args)
+  end
```

Comment:
I mean that full permission is assumed/applied if this method does/returns nothing (passing the access control by default). If a subclass forgets to call super or does not call `apply_gated_discovery`, then the base rules would not be applied. This may not be a concern.

The other approach is a chain of command where if no specific access is applied in the chain, access is rejected.

I suppose, though, that this inversion would be challenging to implement. It'd be translating between selection (grant) and filtering (deny) -- nontrivial.